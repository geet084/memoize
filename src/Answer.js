import React, { Component } from 'react';
import './CSS/Main.scss';

export default class Answer extends Component {
  
  showGuessResult = (answer, count) => {  
    if (answer && count > 0) {
      return "THAT'S CORRECT!";
    } else if (answer === false && count > 0) {
      return "THAT'S INCORRECT!";
    }
  }
  
  render() {
    let { answer, count } = this.props;
    let questionAnswered = answer || answer === false;

    if (questionAnswered) {
      let result = this.showGuessResult(answer, count).includes('INCORRECT');
      let isCorrect;
      result ? isCorrect = 'incorrect' : isCorrect = 'correct'; 
      
      return (
        <div className="answer">
          <button onClick={this.props.nextQuestion}
                  className="next-question">
            Submit answer & show next Question
          </button>
          <p className={isCorrect}>
            {this.showGuessResult(answer, count)}
          </p>
        </div>
      )
    } else {
      return null
    }
  }
}


