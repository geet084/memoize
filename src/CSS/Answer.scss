@mixin colorSet($color, $bgc) {
  color: $color;
  background-color: $bgc;
}

@mixin styling($bord-rad, $margin, $width) {
  border-radius: $bord-rad;
  margin: $margin;
  width: $width;
}

.answer {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

.next-question {
  @include styling(10px, 10px, 290px);
  @include colorSet(#fff, #016FB9);
  font-size: 1rem;
  height: 35px;
  &:hover {
    @include colorSet(#013309, #add9ec);
  }
}

.correct {
  @include colorSet(#fff, green);
  @include styling(15px, 10px 5px 5px 5px, 18.5vw);
  padding: 10px;
}

.incorrect {
  @include colorSet(#fff, maroon);
  @include styling(15px, 10px 5px 5px 5px, 18.5vw);
  padding: 10px;
}