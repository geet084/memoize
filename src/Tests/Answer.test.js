import React from 'react';
import Answer from '../Answer.js';
import { shallow } from 'enzyme';

const mockQuestion = jest.fn();
const mockAnswer = true;
const mockCount = 1;

describe('Answer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Answer nextQuestion={mockQuestion}
              answer={mockAnswer}
              count={mockCount} />
    )
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should register a click to advance to the next question', () => { 
    wrapper.find('.next-question').simulate('click')
    expect(mockQuestion).toBeCalled()
  })

  it('should not render if the user has not tried to answer a question', () => { 
    wrapper = shallow(
      <Answer nextQuestion={mockQuestion}
        answer={undefined}
        count={0} />
    )

    expect(wrapper.html()).toBeNull();
  })

  it('should show whether or not you guessed right', () => {
    
    let result = wrapper.instance().showGuessResult(true, mockCount)
    expect(result).toEqual("THAT'S CORRECT!")

    result = wrapper.instance().showGuessResult(false, mockCount)
    expect(result).toEqual("THAT'S INCORRECT!")
  })
});

