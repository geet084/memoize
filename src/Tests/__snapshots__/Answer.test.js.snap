// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer should match the snapshot with all data passed in correctly 1`] = `
<div>
  <button
    className="next-question"
    onClick={[MockFunction]}
  >
    Click to Submit answer and change to next Question
  </button>
  <p />
</div>
`;
